### PR TITLE
Task/schedules api

### PR DESCRIPTION
**Pull request by scokljat**
## Changes
-Add function for format date with date-fns library.
-Add schedule services,actions and mutations  for fetching, creating, editing and deleting schedules. 
-Add action and mutation for fetching all messages.

## Reference 
https://trello.com/c/rhzMld7v/74-task-task-schedulesapi

### CODE REVIEW

File: src/components/ModalCreateSchedule.vue [hunk 1]
```diff
@@ -41,15 +58,75 @@
...
+    this.fetchAllMessages();
+    if (this.scheduleId > 0 && this.getFormAction === 'update') {
+      const schedule = await this.fetchSchedule(this.scheduleId);
+      this.id = schedule.message.messageId;
+      this.interval = 'Every ' + schedule.schedulerInterval.toLowerCase();
+      this.runDate = formatDate(schedule.runDate, 'yyyy-MM-dd');
+      this.channel = schedule.channel;
+      this.repeat = schedule.repeat;
+      if (schedule.active === 'Active') this.active = true;
+      else this.active = false;
```
Comment (thedev966) [inline src/components/ModalCreateSchedule.vue:104]: Use `this.active = schedule.active === 'Active ?  true : false` instead.

**Review by thedev966 — APPROVED** [review]

File: src/pages/SchedulesPage.vue [hunk 2]
```diff
@@ -143,32 +56,41 @@ export default {
...
+    ...mapGetters({ getSchedules: 'getSchedules' }),
     ...mapGetters({ getFormAction: 'getFormAction' }),
     modalTitle() {
       return this.getFormAction === 'create'
         ? 'Create Schedule'
         : 'Update Schedule';
     },
   },
+  mounted() {
+    this.fetchSchedules({ pageNumber: 1 });
```
Comment (josip2312) [inline src/pages/SchedulesPage.vue:76]: Use vuex bound value here
```suggestion
    this.fetchSchedules({ pageNumber: pagination.page });
```

File: src/store/index.js [hunk 3]
```diff
@@ -43,9 +55,32 @@ export default createStore({
...
+    [SET_SCHEDULES]: (state, payload) => {
+      state.schedules = payload;
+    },
+    [REMOVE_SCHEDULE]: (state, id) => {
+      state.schedules = state.schedules.filter((schedule) => {
+        return schedule.scheduleId !== id;
+      });
+    },
+    [UPDATE_SCHEDULE]: (state, { id, updatedSchedule }) => {
+      const scheduleId = state.schedules.findIndex((schedule) => {
```
Comment (josip2312) [inline src/store/index.js:79]: This is not the schedule id, you're using the `findIndex` method

File: src/store/index.js [hunk 4]
```diff
@@ -61,6 +96,58 @@ export default createStore({
...
+      await SchedulesService.createSchedule(schedule);
+      commit(CLOSE_APP_MODAL);
+    },
+    fetchSchedule({ state }, id) {
+      const schedule = state.schedules.filter((schedule) => {
+        return schedule.scheduleId === id;
+      });
+      return schedule[0];
+    },
+    async editSchedule({ commit }, { id, schedule }) {
```
Comment (josip2312) [inline src/store/index.js:139]: You already have the id inside the schedule object

also update related logic
```suggestion
    async editSchedule({ commit }, schedule) {
```

File: src/store/index.js [hunk 5]
```diff
@@ -61,6 +96,58 @@ export default createStore({
...
+      const updatedSchedule = await SchedulesService.editSchedule(id, schedule);
+      updatedSchedule.nextRun = FormatUtils.formatDate(
+        updatedSchedule.nextRun,
+        'dd MMM yyyy'
+      );
+      updatedSchedule.activeLabel = updatedSchedule.active
+        ? 'Active'
+        : 'Inactive';
+      commit(UPDATE_SCHEDULE, { id, updatedSchedule });
```
Comment (josip2312) [inline src/store/index.js:148]: In addition from above
```suggestion
      commit(UPDATE_SCHEDULE, updatedSchedule);
```

File: src/store/index.js [hunk 6]
```diff
@@ -61,6 +96,58 @@ export default createStore({
...
+      commit(SET_PAGINATION, {
+        page: response.pageable.pageNumber + 1,
+        total: response.totalElements,
+      });
+      commit(SET_SCHEDULES, schedules);
+    },
+    async deleteSchedule({ commit }, id) {
+      await SchedulesService.deleteSchedule(id);
+
+      // set pagination
```
Comment (josip2312) [inline src/store/index.js:125]:
```suggestion
```

File: src/store/index.js [hunk 7]
```diff
@@ -61,6 +96,58 @@ export default createStore({
...
+      const schedules = response.content;
+      schedules.forEach((schedule) => {
+        schedule.nextRun = FormatUtils.formatDate(
+          schedule.nextRun,
+          'dd MMM yyyy'
+        );
+        schedule.activeLabel = schedule.active ? 'Active' : 'Inactive';
+      });
+
+      // set pagination
```
Comment (josip2312) [inline src/store/index.js:115]:
```suggestion
```

File: src/store/index.js (hunk 6 above)
Comment (josip2312) [inline src/store/index.js:125]:
```suggestion
```

**Review by josip2312 — CHANGES_REQUESTED** [review]
Here's the first round :smile:

File: src/components/ModalCreateSchedule.vue [hunk 8]
```diff
@@ -1,55 +1,142 @@
...
+    filterMessages() {
+      const filteredMessages = [];
+      this.messages.forEach((message) => {
+        filteredMessages.push({
+          id: message.messageId,
+          value: message.messageId,
+          label: message.text,
+        });
+      });
+      return filteredMessages;
```
Comment (josip2312) [inline src/components/ModalCreateSchedule.vue:97]: Use map here

Comment (josip2312) [inline src/components/ModalCreateSchedule.vue:97]: Use map here

File: src/store/index.js [hunk 9]
```diff
@@ -3,49 +3,97 @@ import {
...
+        'dd MMM yyyy'
+      );
+      schedule.activeLabel = schedule.active ? 'Active' : 'Inactive';
+    });
+  } else {
+    response.nextRun = FormatUtils.formatDate(response.nextRun, 'dd MMM yyyy');
+    response.activeLabel = response.active ? 'Active' : 'Inactive';
+  }
+  return response;
+}
```
Comment (josip2312) [inline src/store/index.js:34]: Pass an array here from the editSchedule action to avoid these checks

Comment (josip2312) [inline src/store/index.js:34]: Pass an array here from the editSchedule action to avoid these checks

**Review by josip2312 — CHANGES_REQUESTED** [review]

**Review by josip2312 — APPROVED** [review]

**Review by thedev966 — APPROVED** [review]

**Pull request merged**